Assets/HoloToolkit/Input/Scripts/HandsManager.cs

```diff
@@ -21,59 +29,142 @@ public bool HandDetected
+        /// Valid only if a manipulation gesture is in progress.
+        /// </summary>
+        public Vector3 ManipulationHandPosition
+        {
+            get
+            {
+                Vector3 handPosition;
+                if (!currentHandState.properties.location.TryGetPosition(out handPosition))
+                {
+                    handPosition = Vector3.zero;
+                }
+                return handPosition;
+            }
+        }
+
+        private InteractionSourceState currentHandState;
 
         private HashSet<uint> trackedHands = new HashSet<uint>();
 
-        private void Awake()
+        private HashSet<uint> pressedHands = new HashSet<uint>();
+
+        private void OnEnable()
         {
             InteractionManager.SourceDetected += InteractionManager_SourceDetected;
+            InteractionManager.SourcePressed += InteractionManager_SourcePressed;
+            InteractionManager.SourceReleased += InteractionManager_SourceReleased;
+            InteractionManager.SourceUpdated += InteractionManager_SourceUpdated;
             InteractionManager.SourceLost += InteractionManager_SourceLost;
         }
 
+        private void OnDisable()
+        {
+            InteractionManager.SourcePressed -= InteractionManager_SourcePressed;
+            InteractionManager.SourceReleased -= InteractionManager_SourceReleased;
+            InteractionManager.SourceUpdated -= InteractionManager_SourceUpdated;
+            InteractionManager.SourceLost -= InteractionManager_SourceLost;
+        }
+
+        /// <summary>
+        /// Thrown when we detect a hand.
+        /// </summary>
+        /// <param name="state"></param>
         private void InteractionManager_SourceDetected(InteractionSourceState state)
         {
             // Check to see that the source is a hand.
-            if (state.source.kind != InteractionSourceKind.Hand)
+            if (state.source.kind == InteractionSourceKind.Hand)
             {
-                return;
-            }
+                trackedHands.Add(state.source.id);
 
-            trackedHands.Add(state.source.id);
+                if (HandInView != null)
+                {
+                    HandInView(HandDetected);
+                }
+            }
+        }
 
-            if (HandInView != null)
+        /// <summary>
+        /// Thrown when the Interaction Source is pressed.
+        /// </summary>
+        /// <param name="state">The current state of the Interaction source.</param>
+        private void InteractionManager_SourcePressed(InteractionSourceState state)
+        {
+            if (state.source.kind == InteractionSourceKind.Hand)
             {
-                HandInView(HandDetected);
+                if (trackedHands.Contains(state.source.id))
+                {
+                    currentHandState = state;
+
+                    if(pressedHands.Contains(state.source.id))
```

Comment:
another reversed logic on the if (list.contains(foo)){list.add(foo);} rather than if(!list.contains(foo)...